### PR TITLE
DAOS-4694 test: daos object query

### PR DESCRIPTION
**Pull request by shimizukko**
This test partially covers SRS-12-0624.
daos utility supports all command options except rollback, verify,
list-keys and dump.

Write an object with different object classes, query it with daos object
query, and verify the output.

Test-tag-hw-small: pr,hw,small daos_object_query
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/daos_utils.py [hunk 1]
```diff
@@ -584,3 +584,63 @@ def container_list_snaps(self, pool, cont, svc=None):
...
+        data["layout"] = []
+        for i in range(1, len(vals)):
+            if vals[i][3] == "":
+                if "replica" in data["layout"][-1]:
+                    data["layout"][-1]["replica"].append(
+                        (vals[i][4], vals[i][5]))
+                else:
+                    data["layout"][-1]["replica"] = [(vals[i][4], vals[i][5])]
+            else:
+                data["layout"].append({"grp": vals[i][3]})
```
Comment (phender) [inline src/tests/ftest/util/daos_utils.py:644]: Shouldn't we catch any `IndexError` that might be thrown here to allow us to provide a more detailed explanation of what went wrong and possibly the resulting regex result `vals` for quicker debug.

File: src/tests/ftest/control/daos_object_query.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            obj = self.container.written_data[-1].obj
+            obj.close()
+            expected_oid_hi = obj.c_oid.hi
+            expected_oid_lo = obj.c_oid.lo
+            self.log.info("oid.hi = %s", expected_oid_hi)
+            self.log.info("oid.lo = %s", expected_oid_lo)
+            oid_concat = "{}.{}".format(expected_oid_hi, expected_oid_lo)
+            kwargs = {
+                "pool": self.pool.uuid,
+                "svc": self.pool.svc_ranks,
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply src/tests/ftest/control/daos_object_query.py]: Done. Thanks for suggestion.

File: src/tests/ftest/util/daos_utils.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/util/daos_utils.py]: Done

**Review by phender — APPROVED** [review]
LGTM

**Review by amjustin13 — APPROVED** [review]
LGTM

**Comment by shimizukko** [comment]
Functional_Hardware_Large / 1-./nvme/nvme_object.py:NvmeObject.test_nvme_object_single_pool;transport_config-container-transport_config-hosts-createmode-createset-createsize-createsvc-servers-transport_config-timeouts-a31a – NvmeObject -> DAOS-4943
Functional_Hardware_Large / 1-./daos_test/daos_core_test_dfs.py:DaosCoreTestDfs.test_subtest;transport_config-test_F-num_clients-transport_config-hosts-pool-0-1-transport_config-71bd – DaosCoreTestDfs -> DAOS-5215
Functional_Hardware_Large / 1-./aggregation/aggregation_basic.py:DaosAggregationBasic.test_basic_aggregation;container-hosts-np_12-iorflags-oclass_SX-pool-servers-a5f6 – DaosAggregationBasic -> DAOS-5610
Functional_Hardware_Large / 2-./aggregation/aggregation_basic.py:DaosAggregationBasic.test_basic_aggregation;container-hosts-np_12-iorflags-oclass_RP_2GX-pool-servers-e85a – DaosAggregationBasic -> DAOS-5610

**Comment by shimizukko** [comment]
Test Failures:
Functional_Hardware_Medium / 01-./daos_test/daos_core_test-rebuild.py:DaosCoreTestRebuild.test_rebuild;transport_config-test_r_0-10-num_clients-num_replicas-transport_config-hosts-pool-0-1-transport_config-5c97 – DaosCoreTestRebuild -> DAOS-5784

**Pull request merged**